### PR TITLE
docs/sapphire: Add precompile descriptions

### PR DESCRIPTION
**Pull request by jberci**

### CODE REVIEW

**Review by nhynes — APPROVED** [review]
Thanks, this is great. Let's just make sure the gas costs match the time costs. Also, it'll be really great to have this EVM assembly bundled into a `Sapphire.sol` library for people to use.

File: docs/dapp/sapphire/precompiles.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+---
+description: Short guide to using the additional Sapphire precompiles
```
Comment (matevz) [inline docs/dapp/sapphire/precompiles.md]: This is rendered in the doc cards so it needs to be short and concise, like:

```suggestion
description: Additional Sapphire precompiles for encryption and confidentiality
```

File: docs/dapp/sapphire/precompiles.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+---
+
+# Precompiles Available on Sapphire
+
+In addition to the standard EVM precompiles, Sapphire provides three
+further cryptography-related ones to make some operations easier and
+cheaper to perform: x25519 key derivation, and DeoxysII-based encryption
+and decryption.
+
+These can be called in much the same way as other precompiles by
```
Comment (matevz) [inline docs/dapp/sapphire/precompiles.md]:
```suggestion
These can be called in the same way as other precompiles by
```

File: docs/dapp/sapphire/precompiles.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+dispatching calls to specific well-known contract addresses, as
+described below. Input parameters should be packed into a contiguous
+memory region, with each piece of data padded as usual to 32 bytes.
```
Comment (matevz) [inline docs/dapp/sapphire/precompiles.md]:
```suggestion
memory region, with each chunk of data padded to 32 bytes as usual.
```

File: docs/dapp/sapphire/precompiles.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        result := out
+    }
+}
+```
+
+## Library
+
+The examples above show how to call the precompiles directly. For a more
+comfortable experience, wrappers are also provided in the Sapphire
+repository, in `library/Sapphire.sol`. The wrappers have the same
```
Comment (matevz) [inline docs/dapp/sapphire/precompiles.md]: Where is this Sapphire repo? I cannot seem to find any solidity code in https://github.com/oasisprotocol/sapphire-paratime/. Please add a link to `library/Sapphire.sol` file.

**Review by matevz — APPROVED** [review]
Some minor details. However, this PR needs to wait until x25519 and deoxys actually land on Testnet right?

File: docs/dapp/sapphire/precompiles.md (hunk 4 above)
Comment (kostko) [reply docs/dapp/sapphire/precompiles.md]: https://github.com/oasisprotocol/sapphire-paratime/pull/44

Comment (jberci) [reply docs/dapp/sapphire/precompiles.md]: I'll edit the text here once that PR is merged and the path is final.

**Comment by kostko** [comment]
We should probably merge now that the Testnet has been updated to include these.

**Pull request merged**